### PR TITLE
Feature/campaign code types

### PR DESCRIPTION
**Pull request by cpiehl1**
# Jira Issue: [MX-251]

## What?
- Add code types to campaign codes

## Why?
- To not have to manage this separately

## Optional screenshots

https://user-images.githubusercontent.com/35833651/131354923-5029e7be-077c-4502-bf00-f642477074ad.mov

## Optional checklist
- [x] Updated `src/changelog.ts`
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

### CODE REVIEW

File: src/features/tools/campaign-codes/utils.tsx [hunk 1]
```diff
@@ -169,3 +170,17 @@ export const getValidity = (campaign: VoucherCampaign) => {
...
+  return (
+    codeTypes.map((value, index) => {
+      return {
+        key: index + 1,
+        value,
+        label: (value as string).replace(/_/g, ' '),
+      }
+    }) ?? []
+  )
+}
```
Comment (rasmusguterstam) [inline src/features/tools/campaign-codes/utils.tsx:186]: Really nice!!

**Review by rasmusguterstam — CHANGES_REQUESTED** [review]
Let's gooo! The only thing I'm thinking is whether you could Camel Case the enums, e.g. "PROPERTY PARTNER" -> "Property Partner"?

**Comment by cpiehl1** [comment]
@rasmusguterstam Done!

**Review by rasmusguterstam — APPROVED** [review]

File: src/features/tools/campaign-codes/components/CampaignCodeTable.tsx [hunk 2]
```diff
@@ -37,45 +48,86 @@ export const CampaignCodeTable: React.FC<{ filter: CampaignFilter }> = ({
...
 
               return (
-                <Table.Row key={campaign.id}>
+                <Table.Row key={id}>
                   <Table.Cell textAlign="center">
                     {getValidity(campaign)}
                   </Table.Cell>
-                  <Table.Cell>{campaignCode}</Table.Cell>
-                  <Table.Cell>
+                  <Table.Cell textAlign={'center'}>{campaignCode}</Table.Cell>
```
Comment (joacimastrom) [inline src/features/tools/campaign-codes/components/CampaignCodeTable.tsx:89]: I'd make a styled component for this 
```
const CenteredCell = styled(Table.Cell)`
    text-align: center;
`
```

File: src/features/tools/campaign-codes/utils.tsx [hunk 3]
```diff
@@ -169,3 +171,20 @@ export const getValidity = (campaign: VoucherCampaign) => {
...
+
+export const getCodeTypeOptions = () => {
+  const codeTypesQuery = useAvailableCampaignCodeTypesQuery()
+  const codeTypes = codeTypesQuery.data?.availableCampaignCodeTypes ?? []
+  return (
+    codeTypes.map((value, index) => {
+      return {
+        key: index + 1,
+        value,
+        label: (value as string)
```
Comment (joacimastrom) [inline src/features/tools/campaign-codes/utils.tsx:183]:
```suggestion
        label: (value as string).replace("_", " ").toUpperCase()
```

**Review by joacimastrom — APPROVED** [review]
Looks good, well done!

File: src/features/tools/campaign-codes/components/CampaignCodeTable.tsx (hunk 2 above)
Comment (cpiehl1) [reply src/features/tools/campaign-codes/components/CampaignCodeTable.tsx:89]: Good point!

**Pull request merged**